### PR TITLE
Fix saving alignment

### LINKED ISSUE #938: Loss of alignment after saving
## Read and complete the full issue template

- [x] Bug

**Version of ClosedXML**

0.93.0

**What is the current behavior?**

I have an excel-template with horizontal alignment = center in cell "Field1" 

image

I've just opened and saved file:
image

Horizontal alignment has been changed.
If I remove the first column, alignment will not be changed.

**What is the expected behavior or new feature?**

Alignment shouldn't be changed

**Did this work in previous versions of our tool?  Which versions?**

Yes/v0.92.1

## Reproducibility

**Code to reproduce problem:**
```c#
XLWorkbook Wb = new XLWorkbook(FileName)
Wb.SaveAs(FileName);
```
[result.xlsx](https://github.com/ClosedXML/ClosedXML/files/2158974/result.xlsx)
[originalFile.xlsx](https://github.com/ClosedXML/ClosedXML/files/2158975/originalFile.xlsx)

### PR DESCRIPTION
**Pull request by Pankraty**
This PR fixes the regression related to lost alignment on saving.

Depends on #950 (as they both fix the same reference file).

Fixes #938, #940

### CODE REVIEW

**Comment by igitur** [comment]
Please rebase

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 1]
```diff
@@ -3405,7 +3405,7 @@ private void GenerateWorkbookStylesPartContent(WorkbookStylesPart workbookStyles
                 foreach (CellFormat f in workbookStylesPart.Stylesheet.CellFormats)
                 {
                     styleId++;
-                    if (CellFormatsAreEqual(f, ss.Value))
+                    if (CellFormatsAreEqual(f, ss.Value, true))
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Save.cs]: For boolean parameters, I like naming the parameter. It improves readability.

So `if (CellFormatsAreEqual(f, ss.Value, compareAlignment: true))` , please? Sorry to be pedantic.

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 2]
```diff
@@ -3672,7 +3672,17 @@ private static Protection GetProtection(StyleInfo styleInfo)
             };
         }
 
-        private static bool CellFormatsAreEqual(CellFormat f, StyleInfo styleInfo)
+        /// <summary>
+        /// Check if two style are equivalent.
+        /// </summary>
+        /// <param name="f">Style in the OpenXML format.</param>
+        /// <param name="styleInfo">Style in the ClosedXML format.</param>
+        /// <param name="compareAlignment">Flag specifying whether or not compare the alignments of two styles.
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Save.cs]: Any other properties that act in this way, besides alignment?

File: ClosedXML/Excel/XLWorkbook_Save.cs (hunk 1 above)
Comment (Pankraty) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: Agree

File: ClosedXML/Excel/XLWorkbook_Save.cs (hunk 2 above)
Comment (Pankraty) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: From what I know, other properties acts the same way in `x:cellStyleXfs` and `x:cellXfs`.

File: ClosedXML/Excel/XLWorkbook_Save.cs (hunk 1 above)
Comment (Pankraty) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: I have amended the commit

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 3]
```diff
@@ -3572,7 +3572,7 @@ private static void ResolveRest(WorkbookStylesPart workbookStylesPart, SaveConte
             {
                 var info = styleInfo;
                 var foundOne =
-                    workbookStylesPart.Stylesheet.CellFormats.Cast<CellFormat>().Any(f => CellFormatsAreEqual(f, info));
+                    workbookStylesPart.Stylesheet.CellFormats.Cast<CellFormat>().Any(f => CellFormatsAreEqual(f, info, true));
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Save.cs]: Here too.

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 4]
```diff
@@ -3611,7 +3611,7 @@ private static void ResolveRest(WorkbookStylesPart workbookStylesPart, SaveConte
                 var info = styleInfo;
                 var foundOne =
                     workbookStylesPart.Stylesheet.CellStyleFormats.Cast<CellFormat>().Any(
-                        f => CellFormatsAreEqual(f, info));
+                        f => CellFormatsAreEqual(f, info, false));
```
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Save.cs]: Here too.

**Comment by Pankraty** [comment]
Amended once more

**Pull request merged**